intro-OpenStack/ch_HeatTemplates.tex

```diff
@@ -0,0 +1,290 @@
+collectively managed resources is called a \gls{stack}.
+
+\gls{Heat} has its own dashboard interface, which you can find under
+the \strong{Orchestration} tab of the main OpenStack dashboard.  Official
+documentation for Heat and its dashboard interface can be found at the
+following locations:
+\begin{itemize}
+\item \url{https://docs.openstack.org/heat/\osversion}
+\item \url{https://docs.openstack.org/heat-dashboard/\osversion}
+\end{itemize}
+
+\section{\gls{Heat Orchestration Template}s}\label{sec:glsh-orch-templ}
+A \gls{stack}'s resources and their mutual dependencies can be
+specified in a text file, called a \gls{Heat Orchestration Template}
+(\textsc{hot}).  The syntax of these templates conforms to the
+\gls{yaml} standard, for which many text editors provide specialized
+editing modes.  The following example describes a stack consisting of
+a single VM:
+\begin{code}{}
+heat_template_version: 2018-03-02
+
+description: Deploy a single compute instance
+
+parameters:
+  user_network:
+    type: string
+    label: user_network
+    description: Add the required VM network
+    constraints: [ custom_constraint: neutron.network ]
+  user_key:
+    type: string
+    label: ssh_user_key
+    description: Public ssh key for user authentication
+    constraints: [ custom_constraint: nova.keypair ]
+
+resources:
+  my_instance:
+    type: OS::Nova::Server
+    properties:
+      security_groups: [ default ]
+      networks: [ network: { get_param: user_network } ]
+      key_name: { get_param: user_key }
+      image: Ubuntu_16.04_2NICs
+      flavor: m1.small
+\end{code}
+
+Our example contains four main sections:
+\begin{description}
+\item[\texttt{heat\_template\_version}] The \textsc{hot} specification
+  has evolved since its initial release.  The key
+  \lstinline{heat_template_version} indicates the version of the
+  syntax used in this template.  It's value can be a release date or
+  (in recent version) the name of the version.
+\item[\texttt{description}] A description is optional, but
+  recommended.
+\item[\texttt{parameters}] An optional section, \lstinline{parameters}
+  allow users to configure various properties when instantiating a new
+  stack, without having to edit the template itself.  A parameter
+  value can be used elsewhere in the template using the function
+  \lstinline{get_param}.
+\item[\texttt{resources}] This section contains all the resources used
+  by the Stack.  In this case, there is just a single VM instance.
+\end{description}
+Optional additional sections are \strong{\lstinline{parameter_groups}},
+\strong{\lstinline{outputs}}, and \strong{\lstinline{conditions}}.
+
+The
+`\href{https://docs.openstack.org/heat/\osversion/template_guide}{Template
+  Guide}' in the Heat documentation contains a specification of the
+\textsc{hot} format, as well as information on how to describe the
+various types of resources in a template.  \textsc{vsc} also provides
```

Comment:
Another interesting link for the users is https://docs.openstack.org/heat/latest/template_guide/software_deployment.html
If they want to deploy their own automated software deployments with Heat